server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -353,10 +362,15 @@ protected final void innerWriteTo(StreamOutput out, boolean asKey) throws IOExce
         }
         out.writeOptionalWriteable(scroll);
         out.writeOptionalWriteable(source);
-        if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0)) {
+        if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0)
+            && out.getTransportVersion().before(TransportVersion.V_8_500_999)) {
+            List<QueryBuilder> rankQueryBuilders = new ArrayList<>();
+            if (source != null && source.rankBuilder() != null && source.subSearches().size() >= 2) {
+                for (SubSearchSourceBuilder subSearchSourceBuilder : source.subSearches()) {
+                    rankQueryBuilders.add(subSearchSourceBuilder.getQueryBuilder());
+                }
+            }
             out.writeNamedWriteableList(rankQueryBuilders);
-        } else if (rankQueryBuilders.isEmpty() == false) {
-            throw new IllegalArgumentException("cannot serialize [rank] to version [" + out.getTransportVersion() + "]");
```

Comment:
did you mean to delete this?

Reply:
Yes. This logic was moved to `SearchSourceBuilder` instead since where that's where `subSearches` lives.